indico/modules/categories/models/categories.py

```diff
@@ -486,6 +486,12 @@ def visible_categories_query(self):
         cte_query = Category.get_visible_categories_cte(self.id)
         return Category.query.join(cte_query, Category.id == cte_query.c.id)
 
+    def get_hidden_events(self, user=None):
+        """Get all hidden events within the given category and user."""
+        from indico.modules.events import Event
+        hidden_events = Event.query.with_parent(self).filter_by(visibility=0)
```

Comment:
I'd name this `query` or call `.all()` on it (having a query object in something not named query is a bit strange)

Reply:
Yep, added .all()